chore(starters): add gatsby-typescript-eslint-prettier-starter

Tielem opened the pull request:
## Description

This PR adds an adaptation of the default Gatsby starter with TypeScript, ESlint and Prettier added and pre-configured, bringing you everything you need to get up and running with Gatsby in a type-safe and style-safe way.

The difference with existing starters is that they use `gatsby-plugin-typescript`, which Babel transpiles and does not throws type errors during `gatsby develop` and `gatsby build`, significantly reducing the type safety of the project. This starter used `gatsby-plugin-ts`, which in my opinion is the superior TypeScript plugin.

gillkyle commented on docs/starters.yml:
```diff
@@ -6301,3 +6301,19 @@
+- url: https://gatsby-typescript-eslint-prettier-starter.netlify.app/
+  repo: https://github.com/Tielem/gatsby-typescript-eslint-prettier-starter
+  description: This Gatsby starter is an adaptation of the default Gatsby starter with TypeScript, ESlint and Prettier added and pre-configured, bringing you everything you need to get up and running with Gatsby in a type-safe and style-safe way.
+  tags:
```
There is a list of available tags for starters here: https://github.com/gatsbyjs/gatsby/blob/master/docs/categories.yml Could you update these to remove the ones not included in there under `starters`? The features bullet points can have whatever text you'd like, but the tags are restricted so we keep a smaller list to filter by in the showcase.

Tielem commented:
I cleaned up the tags, added a bit more description and added a reference to the Gatsby CLI to the README.md

Is there anything else which you would change?
